### PR TITLE
Show all locations for custom lists in edit locations and detail view

### PR DESCRIPTION
**Pull request by Pururun**
This is to avoid unexpected behavior. The locations are not shown in the select location screen, to align ux with desktop.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)

### CODE REVIEW

**Review by albin-mullvad — COMMENTED** [review]
> Reviewed 8 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 8 of 14 files reviewed, 1 unresolved discussion (waiting on @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/CustomListLocationsViewModel.kt` line 40 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG8Do1Q0SBr3KqNIK:-NvbG8Do1Q0SBr3KqNIL:b-807afe) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/CustomListLocationsViewModel.kt#L40)):*
> ```Kotlin
> 
>     val uiState =
>         combine(relayListUseCase.relatListAll(), _searchTerm, _selectedLocations) {
> ```

Typo?



_Code quote:_
```Kotlin
relat
```

**Review by albin-mullvad — CHANGES_REQUESTED** [review]
> Reviewed 1 of 14 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 9 of 14 files reviewed, 3 unresolved discussions (waiting on @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 61 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbGWMU3gCbZNxOOUUD:-NvbGWMV1_bPEWGe2GJF:b3b5cm2) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L61)):*
> ```Kotlin
>     fun selectedRelayItem(): Flow<RelayItem?> = relayListWithSelection().map { it.selectedItem }
> 
>     fun relayListFiltered(): Flow<List<RelayItem.Country>> =
> ```

I suggest: `filteredRelayList`



_Code quote:_
```Kotlin
relayListFiltered
```
___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 64 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG_iV8i_8XQgW5dW2:-NvbG_iV8i_8XQgW5dW3:btfporh) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L64)):*
> ```Kotlin
>         relayListWithSelection().map { it.filteredCountries }
> 
>     fun relatListAll(): Flow<List<RelayItem.Country>> =
> ```

I suggest something like: `unfilteredRelayList` or `fullRelayList`



_Code quote:_
```Kotlin
relatListAll
```

**Review by albin-mullvad — COMMENTED** [review]
> Reviewed 1 of 14 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 10 of 14 files reviewed, 3 unresolved discussions (waiting on @Pururun)

**Review by Rawa — APPROVED** [review]
> Reviewed 10 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: all files reviewed, 3 unresolved discussions (waiting on @albin-mullvad and @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 64 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG_iV8i_8XQgW5dW2:-NvbHIt_-nzRH5Ers1Es:brv5p6x) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L64)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

I suggest something like: `unfilteredRelayList` or `fullRelayList`
</blockquote></details>

I agree the naming can be better, personally I think `relayList` and `filteredRelayList` would be clear enough.

**Review by albin-mullvad — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: all files reviewed, 3 unresolved discussions (waiting on @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 64 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG_iV8i_8XQgW5dW2:-NvbIYVk6zIfa1_NDQaS:b4t7a1d) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L64)):*
<details><summary><i>Previously, Rawa (David Göransson) wrote…</i></summary><blockquote>

I agree the naming can be better, personally I think `relayList` and `filteredRelayList` would be clear enough.
</blockquote></details>

Yeah, that also works! Was thinking of suggesting that but got the impression that @Pururun might see value in being more explicit. I'm fine either way!

**Review by Pururun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: all files reviewed, 5 unresolved discussions (waiting on @albin-mullvad and @Rawa)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt` line 183 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbPtGC-aWKze8O689E:-NvcnW-U7BumFyEsl65C:b-k1vkhl) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt#L183)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

nit: not needed, right?
</blockquote></details>

Yeah, redid a little bit though since I realized my implementation would include empty cities.
___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt` line 195 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbPK3_35qpvQVlSJ2L:-Nvco-sZAIQ50YA5Yacm:b7ft538) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt#L195)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

Can the readability be improved? Quite hard to follow. Maybe using some helper function or local variable :thinking:
</blockquote></details>

I made it like that to not filter twice, but the previous implementation did also filter twice and was much more readable so I just changed it to use that one instead.
___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 61 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbGWMU3gCbZNxOOUUD:-NvcpBKo8Yry56JIRq6z:b-em7y0d) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L61)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

I suggest: `filteredRelayList`
</blockquote></details>

It was not needed so I just removed it.
___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt` line 64 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG_iV8i_8XQgW5dW2:-NvcpIBs-33rzUty2YRD:b-khwwrh) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/RelayListUseCase.kt#L64)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

Yeah, that also works! Was thinking of suggesting that but got the impression that @Pururun might see value in being more explicit. I'm fine either way!
</blockquote></details>

I would say I prefer `fullRelayList`
___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/CustomListLocationsViewModel.kt` line 40 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbG8Do1Q0SBr3KqNIK:-Nvcp_5dCRGUIGn506k6:b-oob2z9) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/CustomListLocationsViewModel.kt#L40)):*
<details><summary><i>Previously, albin-mullvad wrote…</i></summary><blockquote>

Typo?
</blockquote></details>

Wonder what happened there :thinking: Fixed.

**Review by Pururun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 7 of 14 files reviewed, 3 unresolved discussions (waiting on @albin-mullvad and @Rawa)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/SelectLocationViewModel.kt` line 38 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NveZnjfAWm88uGqWKnP:-NveZnjfAWm88uGqWKnQ:brmm5x6) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1c5b8039cb9e8ee1d420634c56550b8d52510784/android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/SelectLocationViewModel.kt#L38)):*
> ```Kotlin
>     private val _searchTerm = MutableStateFlow(EMPTY_SEARCH_TERM)
> 
>     @Suppress("DestructuringDeclarationWithTooManyEntries")
> ```

Added this since I think that it is better to use a destructuring declaration and we only add an entry here that we don't use anyway.

**Review by Pururun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 7 of 14 files reviewed, 3 unresolved discussions (waiting on @albin-mullvad and @Rawa)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/SelectLocationViewModel.kt` line 38 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NveZnjfAWm88uGqWKnP:-Nve_h_O0a5MPM4L91Ck:b-mzij1q) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1c5b8039cb9e8ee1d420634c56550b8d52510784/android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/SelectLocationViewModel.kt#L38)):*
<details><summary><i>Previously, Pururun (Jonatan Rhodin) wrote…</i></summary><blockquote>

Added this since I think that it is better to use a destructuring declaration and we only add an entry here that we don't use anyway.
</blockquote></details>

It should be noted that this is completely rewritten in the gRPC branch so hopefully this will not be in the code base for long.

**Review by albin-mullvad — COMMENTED** [review]
> Reviewed 3 of 4 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: 13 of 14 files reviewed, all discussions resolved (waiting on @Rawa)

**Review by Rawa — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: all files reviewed, 1 unresolved discussion (waiting on @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt` line 187 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-Nverm1aC1-bIx5PQvUs:-Nverm1aC1-bIx5PQvUt:b-dl717p) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1c5b8039cb9e8ee1d420634c56550b8d52510784/android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt#L187)):*
> ```Kotlin
>                     val relays =
>                         city.relays.filterRelayByOwnershipAndProviders(ownership, providers).map {
>                             it.copy()
> ```

Unnecessary copy?

**Review by albin-mullvad — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: all files reviewed, 1 unresolved discussion (waiting on @Pururun)

___
*[`android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt` line 183 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-NvbPtGC-aWKze8O689E:-Nvf2WhP-fgwmekKXPlg:b-lria0s) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/51e25e56094aa81330ab71bbe69eb12725bc898d/android/app/src/main/kotlin/net/mullvad/mullvadvpn/relaylist/RelayListExtensions.kt#L183)):*
<details><summary><i>Previously, Pururun (Jonatan Rhodin) wrote…</i></summary><blockquote>

Yeah, redid a little bit though since I realized my implementation would include empty cities.
</blockquote></details>

Still has the `this` keyword though? But I guess that's fine if you believe that it improves readability

**Review by Rawa — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by Rawa — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-:-Nvf82EC0Cya_XwrkdcM:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by albin-mullvad — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135#-:-Nvf840uBAxKO_KRD-UE:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by Rawa — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6135)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved
